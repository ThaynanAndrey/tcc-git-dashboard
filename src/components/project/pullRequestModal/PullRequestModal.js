import React from 'react';

import Modal from '../../layout/Modal';
import PullRequestContentModal from './PullRequestContentModal'
import PullRequestFooterModal from './PullRequestFooterModal';

/**
 * Stateless component with the function to presenting the Pull Request Details in Modal.
 * 
 * @author Thaynan Nunes
 */
const PullRequestModal = ({ pullRequest, closePullRequestDetails }) => {
    return (
        <Modal isFixed={true}>
            <PullRequestContentModal
                pullRequest={pullRequest}
            />

            <PullRequestFooterModal
                closePullRequestDetails={closePullRequestDetails}
            />
        </Modal>
    );
}

export default PullRequestModal;