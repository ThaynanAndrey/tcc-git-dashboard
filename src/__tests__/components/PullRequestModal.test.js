import React from 'react';
import { mount, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import PullRequestModal from '../../components/project/pullRequestModal/PullRequestModal';

describe('Test PullRequestModal', () => {
  let tree;
  let pullRequest;
  let closePullRequestDetails;
  
  beforeEach(() => {
    pullRequest = {
        commits: [],
        comments: [],
        description: "nothing",
        responsavel: { nome: "ze" },
        repositorio: { nome: "eh" }
    };
    closePullRequestDetails = jest.fn();

    tree = mount(
        <PullRequestModal
            pullRequest={pullRequest}
            closePullRequestDetails={closePullRequestDetails}
        />
    );
  });

  it('renders correctly a component to show PullRequestModal', () => {
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders correctly a component to show PullRequestModal with correct props', () => {
    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.prop('pullRequest')).toEqual(pullRequest);
    expect(tree.prop('closePullRequestDetails')).toEqual(closePullRequestDetails);
  });
});