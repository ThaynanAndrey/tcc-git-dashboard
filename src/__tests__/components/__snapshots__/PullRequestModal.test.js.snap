// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PullRequestModal renders correctly a component to show PullRequestModal 1`] = `
<PullRequestModal
  closePullRequestDetails={[MockFunction]}
  pullRequest={
    Object {
      "comments": Array [],
      "commits": Array [],
      "description": "nothing",
      "repositorio": Object {
        "nome": "eh",
      },
      "responsavel": Object {
        "nome": "ze",
      },
    }
  }
>
  <Modal
    isFixed={true}
  >
    <div
      className="modal modal-fixed-footer modal-styles"
    >
      <PullRequestContentModal
        pullRequest={
          Object {
            "comments": Array [],
            "commits": Array [],
            "description": "nothing",
            "repositorio": Object {
              "nome": "eh",
            },
            "responsavel": Object {
              "nome": "ze",
            },
          }
        }
      >
        <div
          className="modal-content"
          style={
            Object {
              "backgroundColor": "#FFFAFA",
              "padding": "0",
            }
          }
        >
          <h5
            style={
              Object {
                "margin": "0",
                "padding": "18px",
                "textAlign": "center",
              }
            }
          >
             
             
          </h5>
          <div
            style={
              Object {
                "padding": "0px 18px 18px 18px",
              }
            }
          >
            <div
              className="card"
            >
              <div
                className="card-content"
                style={
                  Object {
                    "paddingBottom": "0",
                    "paddingTop": "10px",
                  }
                }
              >
                <span
                  className="card-title"
                >
                  Detalhes
                </span>
              </div>
              <div
                className="card-action"
                style={
                  Object {
                    "maxHeight": "30vh",
                    "overflow": "auto",
                    "paddingTop": "0",
                  }
                }
              >
                <ul>
                  <li
                    className="row"
                    style={
                      Object {
                        "marginBottom": 0,
                      }
                    }
                  >
                    <p
                      className="col"
                      style={
                        Object {
                          "fontWeight": "bold",
                          "margin": 0,
                          "paddingRight": 0,
                        }
                      }
                    >
                      Status:
                    </p>
                    <p
                      className="col"
                      style={
                        Object {
                          "margin": 0,
                        }
                      }
                    >
                       
                       
                    </p>
                  </li>
                  <li
                    className="row"
                    style={
                      Object {
                        "marginBottom": 0,
                      }
                    }
                  >
                    <p
                      className="col"
                      style={
                        Object {
                          "fontWeight": "bold",
                          "margin": 0,
                          "paddingRight": 0,
                        }
                      }
                    >
                      Data de criação:
                    </p>
                    <p
                      className="col"
                      style={
                        Object {
                          "margin": 0,
                        }
                      }
                    >
                       
                       
                    </p>
                  </li>
                  <li
                    className="row"
                    style={
                      Object {
                        "marginBottom": 0,
                      }
                    }
                  >
                    <p
                      className="col"
                      style={
                        Object {
                          "fontWeight": "bold",
                          "margin": 0,
                          "paddingRight": 0,
                        }
                      }
                    >
                      Última atualização:
                    </p>
                    <p
                      className="col"
                      style={
                        Object {
                          "margin": 0,
                        }
                      }
                    >
                       
                       
                    </p>
                  </li>
                  <li
                    className="row"
                    style={
                      Object {
                        "marginBottom": 0,
                      }
                    }
                  >
                    <p
                      className="col"
                      style={
                        Object {
                          "fontWeight": "bold",
                          "margin": 0,
                          "paddingRight": 0,
                        }
                      }
                    >
                      Login responsável:
                    </p>
                    <p
                      className="col"
                      style={
                        Object {
                          "margin": 0,
                        }
                      }
                    >
                       
                      ze
                       
                    </p>
                  </li>
                  <li
                    className="row"
                    style={
                      Object {
                        "marginBottom": 0,
                      }
                    }
                  >
                    <p
                      className="col"
                      style={
                        Object {
                          "fontWeight": "bold",
                          "margin": 0,
                          "paddingRight": 0,
                        }
                      }
                    >
                      Repositório:
                    </p>
                    <p
                      className="col"
                      style={
                        Object {
                          "margin": 0,
                        }
                      }
                    >
                       
                      eh
                       
                    </p>
                  </li>
                </ul>
              </div>
            </div>
            <PullRequestDetailsCollapsible
              pullRequest={
                Object {
                  "comments": Array [],
                  "commits": Array [],
                  "description": "nothing",
                  "repositorio": Object {
                    "nome": "eh",
                  },
                  "responsavel": Object {
                    "nome": "ze",
                  },
                }
              }
            >
              <Collapsible>
                <ul
                  className="collapsible"
                >
                  <PullRequestDescriptionCollapsible
                    pullRequest={
                      Object {
                        "comments": Array [],
                        "commits": Array [],
                        "description": "nothing",
                        "repositorio": Object {
                          "nome": "eh",
                        },
                        "responsavel": Object {
                          "nome": "ze",
                        },
                      }
                    }
                  >
                    <CollapsibleCard
                      isActive={false}
                    >
                      <li
                        className=""
                      >
                        <div
                          className="collapsible-header"
                          style={
                            Object {
                              "display": "flex",
                              "justifyContent": "space-between",
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "display": "flex",
                              }
                            }
                          >
                            <i
                              className="material-icons"
                            >
                              dehaze
                            </i>
                            Descrição
                          </div>
                        </div>
                        <div
                          className="collapsible-body body-collapsible-project descriptionPullRequestContent"
                          id="markdown-description"
                        />
                      </li>
                    </CollapsibleCard>
                  </PullRequestDescriptionCollapsible>
                  <PullRequestCommitsCollapsible
                    pullRequest={
                      Object {
                        "comments": Array [],
                        "commits": Array [],
                        "description": "nothing",
                        "repositorio": Object {
                          "nome": "eh",
                        },
                        "responsavel": Object {
                          "nome": "ze",
                        },
                      }
                    }
                  >
                    <CollapsibleCard
                      isActive={false}
                    >
                      <li
                        className=""
                      >
                        <div
                          className="collapsible-header"
                          style={
                            Object {
                              "display": "flex",
                              "justifyContent": "space-between",
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "display": "flex",
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "backgroundColor": "red",
                                  "borderRadius": "0.5rem",
                                  "color": "white",
                                  "height": "27px",
                                  "marginRight": "1rem",
                                  "textAlign": "center",
                                  "width": "2rem",
                                }
                              }
                            >
                              0
                            </div>
                            Commits
                          </div>
                        </div>
                      </li>
                    </CollapsibleCard>
                  </PullRequestCommitsCollapsible>
                  <PullRequestCommentsCollapsible
                    pullRequest={
                      Object {
                        "comments": Array [],
                        "commits": Array [],
                        "description": "nothing",
                        "repositorio": Object {
                          "nome": "eh",
                        },
                        "responsavel": Object {
                          "nome": "ze",
                        },
                      }
                    }
                  >
                    <CollapsibleCard
                      isActive={false}
                    >
                      <li
                        className=""
                      >
                        <div
                          className="collapsible-header"
                          style={
                            Object {
                              "display": "flex",
                              "justifyContent": "space-between",
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "display": "flex",
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "backgroundColor": "red",
                                  "borderRadius": "0.5rem",
                                  "color": "white",
                                  "height": "27px",
                                  "marginRight": "1rem",
                                  "textAlign": "center",
                                  "width": "2rem",
                                }
                              }
                            >
                              0
                            </div>
                            Comentários
                          </div>
                        </div>
                      </li>
                    </CollapsibleCard>
                  </PullRequestCommentsCollapsible>
                </ul>
              </Collapsible>
            </PullRequestDetailsCollapsible>
          </div>
        </div>
      </PullRequestContentModal>
      <PullRequestFooterModal
        closePullRequestDetails={[MockFunction]}
      >
        <div
          className="modal-footer"
        >
          <div
            className="modal-close waves-effect waves-green btn-flat"
            onClick={[Function]}
          >
            Fechar
          </div>
        </div>
      </PullRequestFooterModal>
    </div>
  </Modal>
</PullRequestModal>
`;

exports[`Test PullRequestModal renders correctly a component to show PullRequestModal with correct props 1`] = `
<PullRequestModal
  closePullRequestDetails={[MockFunction]}
  pullRequest={
    Object {
      "comments": Array [],
      "commits": Array [],
      "description": "nothing",
      "repositorio": Object {
        "nome": "eh",
      },
      "responsavel": Object {
        "nome": "ze",
      },
    }
  }
>
  <Modal
    isFixed={true}
  >
    <div
      className="modal modal-fixed-footer modal-styles"
    >
      <PullRequestContentModal
        pullRequest={
          Object {
            "comments": Array [],
            "commits": Array [],
            "description": "nothing",
            "repositorio": Object {
              "nome": "eh",
            },
            "responsavel": Object {
              "nome": "ze",
            },
          }
        }
      >
        <div
          className="modal-content"
          style={
            Object {
              "backgroundColor": "#FFFAFA",
              "padding": "0",
            }
          }
        >
          <h5
            style={
              Object {
                "margin": "0",
                "padding": "18px",
                "textAlign": "center",
              }
            }
          >
             
             
          </h5>
          <div
            style={
              Object {
                "padding": "0px 18px 18px 18px",
              }
            }
          >
            <div
              className="card"
            >
              <div
                className="card-content"
                style={
                  Object {
                    "paddingBottom": "0",
                    "paddingTop": "10px",
                  }
                }
              >
                <span
                  className="card-title"
                >
                  Detalhes
                </span>
              </div>
              <div
                className="card-action"
                style={
                  Object {
                    "maxHeight": "30vh",
                    "overflow": "auto",
                    "paddingTop": "0",
                  }
                }
              >
                <ul>
                  <li
                    className="row"
                    style={
                      Object {
                        "marginBottom": 0,
                      }
                    }
                  >
                    <p
                      className="col"
                      style={
                        Object {
                          "fontWeight": "bold",
                          "margin": 0,
                          "paddingRight": 0,
                        }
                      }
                    >
                      Status:
                    </p>
                    <p
                      className="col"
                      style={
                        Object {
                          "margin": 0,
                        }
                      }
                    >
                       
                       
                    </p>
                  </li>
                  <li
                    className="row"
                    style={
                      Object {
                        "marginBottom": 0,
                      }
                    }
                  >
                    <p
                      className="col"
                      style={
                        Object {
                          "fontWeight": "bold",
                          "margin": 0,
                          "paddingRight": 0,
                        }
                      }
                    >
                      Data de criação:
                    </p>
                    <p
                      className="col"
                      style={
                        Object {
                          "margin": 0,
                        }
                      }
                    >
                       
                       
                    </p>
                  </li>
                  <li
                    className="row"
                    style={
                      Object {
                        "marginBottom": 0,
                      }
                    }
                  >
                    <p
                      className="col"
                      style={
                        Object {
                          "fontWeight": "bold",
                          "margin": 0,
                          "paddingRight": 0,
                        }
                      }
                    >
                      Última atualização:
                    </p>
                    <p
                      className="col"
                      style={
                        Object {
                          "margin": 0,
                        }
                      }
                    >
                       
                       
                    </p>
                  </li>
                  <li
                    className="row"
                    style={
                      Object {
                        "marginBottom": 0,
                      }
                    }
                  >
                    <p
                      className="col"
                      style={
                        Object {
                          "fontWeight": "bold",
                          "margin": 0,
                          "paddingRight": 0,
                        }
                      }
                    >
                      Login responsável:
                    </p>
                    <p
                      className="col"
                      style={
                        Object {
                          "margin": 0,
                        }
                      }
                    >
                       
                      ze
                       
                    </p>
                  </li>
                  <li
                    className="row"
                    style={
                      Object {
                        "marginBottom": 0,
                      }
                    }
                  >
                    <p
                      className="col"
                      style={
                        Object {
                          "fontWeight": "bold",
                          "margin": 0,
                          "paddingRight": 0,
                        }
                      }
                    >
                      Repositório:
                    </p>
                    <p
                      className="col"
                      style={
                        Object {
                          "margin": 0,
                        }
                      }
                    >
                       
                      eh
                       
                    </p>
                  </li>
                </ul>
              </div>
            </div>
            <PullRequestDetailsCollapsible
              pullRequest={
                Object {
                  "comments": Array [],
                  "commits": Array [],
                  "description": "nothing",
                  "repositorio": Object {
                    "nome": "eh",
                  },
                  "responsavel": Object {
                    "nome": "ze",
                  },
                }
              }
            >
              <Collapsible>
                <ul
                  className="collapsible"
                >
                  <PullRequestDescriptionCollapsible
                    pullRequest={
                      Object {
                        "comments": Array [],
                        "commits": Array [],
                        "description": "nothing",
                        "repositorio": Object {
                          "nome": "eh",
                        },
                        "responsavel": Object {
                          "nome": "ze",
                        },
                      }
                    }
                  >
                    <CollapsibleCard
                      isActive={false}
                    >
                      <li
                        className=""
                      >
                        <div
                          className="collapsible-header"
                          style={
                            Object {
                              "display": "flex",
                              "justifyContent": "space-between",
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "display": "flex",
                              }
                            }
                          >
                            <i
                              className="material-icons"
                            >
                              dehaze
                            </i>
                            Descrição
                          </div>
                        </div>
                        <div
                          className="collapsible-body body-collapsible-project descriptionPullRequestContent"
                          id="markdown-description"
                        />
                      </li>
                    </CollapsibleCard>
                  </PullRequestDescriptionCollapsible>
                  <PullRequestCommitsCollapsible
                    pullRequest={
                      Object {
                        "comments": Array [],
                        "commits": Array [],
                        "description": "nothing",
                        "repositorio": Object {
                          "nome": "eh",
                        },
                        "responsavel": Object {
                          "nome": "ze",
                        },
                      }
                    }
                  >
                    <CollapsibleCard
                      isActive={false}
                    >
                      <li
                        className=""
                      >
                        <div
                          className="collapsible-header"
                          style={
                            Object {
                              "display": "flex",
                              "justifyContent": "space-between",
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "display": "flex",
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "backgroundColor": "red",
                                  "borderRadius": "0.5rem",
                                  "color": "white",
                                  "height": "27px",
                                  "marginRight": "1rem",
                                  "textAlign": "center",
                                  "width": "2rem",
                                }
                              }
                            >
                              0
                            </div>
                            Commits
                          </div>
                        </div>
                      </li>
                    </CollapsibleCard>
                  </PullRequestCommitsCollapsible>
                  <PullRequestCommentsCollapsible
                    pullRequest={
                      Object {
                        "comments": Array [],
                        "commits": Array [],
                        "description": "nothing",
                        "repositorio": Object {
                          "nome": "eh",
                        },
                        "responsavel": Object {
                          "nome": "ze",
                        },
                      }
                    }
                  >
                    <CollapsibleCard
                      isActive={false}
                    >
                      <li
                        className=""
                      >
                        <div
                          className="collapsible-header"
                          style={
                            Object {
                              "display": "flex",
                              "justifyContent": "space-between",
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "display": "flex",
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "backgroundColor": "red",
                                  "borderRadius": "0.5rem",
                                  "color": "white",
                                  "height": "27px",
                                  "marginRight": "1rem",
                                  "textAlign": "center",
                                  "width": "2rem",
                                }
                              }
                            >
                              0
                            </div>
                            Comentários
                          </div>
                        </div>
                      </li>
                    </CollapsibleCard>
                  </PullRequestCommentsCollapsible>
                </ul>
              </Collapsible>
            </PullRequestDetailsCollapsible>
          </div>
        </div>
      </PullRequestContentModal>
      <PullRequestFooterModal
        closePullRequestDetails={[MockFunction]}
      >
        <div
          className="modal-footer"
        >
          <div
            className="modal-close waves-effect waves-green btn-flat"
            onClick={[Function]}
          >
            Fechar
          </div>
        </div>
      </PullRequestFooterModal>
    </div>
  </Modal>
</PullRequestModal>
`;
